src/lib/viewers/controls/color-picker/ColorPickerControl.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import ColorPickerPalette from './ColorPickerPalette';
+import { AnnotationMode } from '../annotations/types';
+import './ColorPickerControl.scss';
+
+export type Props = {
+    annotationMode?: AnnotationMode;
+    onAnnotationColorClick: (color: string) => void;
+    isActive?: boolean;
+};
+
+export default function ColorPickerControl({
+    annotationMode,
+    isActive = false,
+    onAnnotationColorClick,
+    ...rest
+}: Props): JSX.Element | null {
+    const [isColorPickerToggled, setIsColorPickerToggled] = useState(false);
+
+    if (annotationMode !== AnnotationMode.DRAWING) {
+        return null;
+    }
+
+    return (
+        <div className="bp-ColorPickerControl">
+            {isColorPickerToggled && (
+                <div className="bp-ColorPickerControl-palette">
+                    <ColorPickerPalette
+                        onColorSelect={(color: string): void => {
+                            setIsColorPickerToggled(false);
+                            onAnnotationColorClick(color);
+                        }}
+                    />
+                </div>
+            )}
+            <button
```

Comment:
Does this need a resin tag and/or testid attributes? Similarly for the `ColorPickerPalette`